File: src/main/java/play/pluv/playlist/application/PlayListConnector.java
package play.pluv.playlist.application;

import java.util.List;
import play.pluv.music.domain.MusicStreaming;
import play.pluv.playlist.domain.PlayList;
import play.pluv.playlist.domain.PlayListId;
import play.pluv.playlist.domain.PlayListMusic;

public interface PlayListConnector {

  List<PlayList> getPlayList(final String authKey);

  List<PlayListMusic> getMusics(final String playListId, final String authKey);

  PlayListId createPlayList(final String authKey, final String name);

  MusicStreaming supportedType();
}

File: src/main/kotlin/play/pluv/playlist/application/PlayListConnector.kt
package play.pluv.playlist.application

import play.pluv.playlist.domain.MusicStreaming
import play.pluv.playlist.domain.PlayList
import play.pluv.playlist.domain.PlayListId
import play.pluv.playlist.domain.PlayListMusic

interface PlayListConnector {
    fun getPlayList(accessToken: String): List<PlayList>

    fun getMusics(playListId: String, accessToken: String): List<PlayListMusic>

    fun createPlayList(accessToken: String, name: String): PlayListId

    fun supportedType(): MusicStreaming
}
